### ISSUE TITLE
Linking issue with libboost and libyaml with CPM

### ISSUE THREAD

**Issue by ttmchiou** [tenstorrent/tt-metal#9532]
**Describe the bug**
Using CPM to install libboost and libyaml does not work. Compilation works but objects are not linked properly against it.

Previously this does not break our CI machines because the deps are installed at the system level.
When running on the latest dev docker image where these deps are not installed, devs will find issues importing libboost or libyaml

**To Reproduce**
```
clone tt-metal
export TT_METAL_HOME=$(pwd)
export ARCH_NAME
export PYTHONPATH=$(pwd)
./build_metal.sh
./create_venv.sh
pytest --co models/demos/resnet/tests/test_perf_resnet_2cqs.py::test_perf_2cqs_bare_metal[20-0.005-16-device_params0]
```

**Expected behavior**
```
mchiou@aus-gs-07-special-mchiou-for-reservation-1578105:~/tt-metal$ pytest --co models/demos/resnet/tests/test_perf_resnet_2cqs.py::test_perf_2cqs_bare_metal[20-0.005-16-device_params0]
=================================================================================================================================================== test session starts ====================================================================================================================================================
platform linux -- Python 3.8.10, pytest-7.2.2, pluggy-1.5.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /home/mchiou/tt-metal, configfile: pytest.ini
plugins: dash-2.15.0, anyio-4.4.0, split-0.8.2, timeout-2.2.0
timeout: 2400.0s
timeout method: signal
timeout func_only: False
collecting ... 2024-06-18 22:24:05.019 | DEBUG    | ttnn:<module>:142 - Initial ttnn.CONFIG:
{'cache_path': PosixPath('/home/mchiou/.cache/ttnn'),
 'comparison_mode_pcc': 0.9999,
 'enable_comparison_mode': False,
 'enable_detailed_buffer_report': False,
 'enable_detailed_tensor_report': False,
 'enable_fast_runtime_mode': True,
 'enable_graph_report': False,
 'enable_logging': False,
 'enable_model_cache': False,
 'model_cache_path': PosixPath('/home/mchiou/.cache/ttnn/models'),
 'report_name': None,
 'root_report_path': PosixPath('generated/ttnn/reports'),
 'throw_exception_on_fallback': False,
 'tmp_dir': PosixPath('/tmp/ttnn')}
collected 1 item

<Module models/demos/resnet/tests/test_perf_resnet_2cqs.py>
  <Function test_perf_2cqs_bare_metal[20-0.005-16-device_params0]>
```

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Version of software (eg. commit)

**Additional context**

**Comment by yan-zaretskiy** [comment]
The issue has to do with boost being built as a static lib on the very first clean build, but as a shared lib for all subsequent builds, due to the CMake variable leaking into the Boost CMake config. I'll push a fix soon to force the build to always be static.

**Comment by ttmchiou** [comment]
closing as issue is fixed in [#9532](https://github.com/tenstorrent/tt-metal/pull/9619)
